### DISCUSSION TITLE
Is build file auto-reload a sensible default?

### DISCUSSION THREAD

**Discussion by susliko** [scalameta/metals#5498] [General]
I've used [rust-tools.nvim](https://github.com/simrat39/rust-tools.nvim) for some time and enjoyed that it picks any changes in `Cargo.toml` automatically.

I thought that this might make nvim-metals UX a little smoother. At the moment it raises a popup upon changes in build files.
Would it be practical at all considering a speed of re-import?

If yes, maybe some user survey could be done here? 

![image](https://github.com/scalameta/nvim-metals/assets/15569000/32b4fcd1-8610-41df-828d-5994c66427bf)

**Comment by ckipp01** [comment]
So there were quite a few discussion about this on the main Metals repo, you could probably still find them. However, the idea was that if we would do an auto-import without any prompting it could lead to a really bad user experience since you'd probably be triggering way more imports (which can be really expensive) than needed. While there might be room to improve this, it's tricky. Either way the change would need to actually be made in upstream Metals, not here. I'll go ahead and transfer this discussion over there.

**Reply by tgodzik** [reply]
I wouldn't do that for sbt + Bloop. We can try that when using sbt via BSP, but we would need to make the experience there better.

**Reply by susliko** [reply]
Got it, thanks!
